iroh-net/src/bin/derper.rs

```diff
@@ -206,23 +151,94 @@ fn load_private_key(filename: impl AsRef<Path>) -> Result<rustls::PrivateKey> {
 
 #[derive(Serialize, Deserialize)]
 struct Config {
+    /// PrivateKey for this Derper.
```

Comment:
General thought, now that we use the config struct, could we make this a little more organized and split between listening details and tls config eg


```rust
struct Config {
    private_key: key::node::SecretKey,
    tls: TlsConfig,
   host: HostConfig,
   limits: Limits,
   derp: DerpConfig,
}

struct HostConfig {
    addr: SocketAddr,
    captive_portal_port: u16,
    stun_port: u16,
    hostname: String,   
    enable_stun: bool,
    enable_derp: bool,
}

struct DerpConfig {
    mesh_psk_file: Option<PathBuf>,
    mesh_with: Vec<Url>,
}

struct TlsConfig {
    cert_mode: CertMode,
    prod_tls: bool,
    tls_contact: Option<String>,
    cert_dir: Option<PathBuf>,
}

struct Limits {
    accept_conn_limit: Option<f64>,
    accept_conn_burst: Option<usize>,
}
```

Reply:
I adjusted this a bit. The main config contains all the "required" information, and the `MeshConfig`, `Limits`, and `TlsConfig` are all optional.

This actually improved the logic of the main function as well, so that is very nice!